wp-cache-sanitizer.php

```diff
@@ -0,0 +1,726 @@
+            }
+            $sanitized_value = "\"$sanitized_value\"";
+            break;
+
+        case 'cached_direct_pages':
+            if ( is_array( $value ) )  {
+                foreach ( $value as $page ) {
+                    $page = str_replace( get_option( 'siteurl' ), '', $page );
+                    if( substr( $page, 0, 1 ) != '/' )
+                        $page = '/' . $page;
+                    $page = esc_sql( $page );
+                    $page = "'$page'";
+                    $sanitized_values[] = $page;
+                }
+                $sanitized_values = array_unique( $sanitized_values );
+                $sanitized_value = 'array(' . implode(', ', $sanitized_values ) . ');';
+            } else {
+                $sanitized_value = '';
+            }
+            break;
+
+        case 'file_prefix':
+            $sanitized_value = ! is_string( $value ) ? 'wp-cache-' : esc_html( $value );
+            $sanitized_value = "\"$sanitized_value\"";
+            break;
+
+        case "wp_cache_debug_ip":
+            $sanitized_value = filter_var( $value, FILTER_VALIDATE_IP ) === false ? "" : esc_html( $value );
+            $sanitized_value = "\"$sanitized_value\"" ;
+            break;
+
+        case "wp_cache_debug_log":
+            $sanitized_value = md5( time() ) . ".txt";
+            $logname = str_replace( '.txt', '', $value );
+            if ( strlen( $logname ) === 32 && ctype_xdigit( $logname ) ) {
+                $sanitized_value = $value;
+            }
+            $sanitized_value = "\"$sanitized_value\"" ;
+            break;
+
+        case "wp_cache_home_path":
+            $sanitized_value = '/';
+            $home_path = parse_url( site_url() );
+            $home_path = trailingslashit( array_key_exists( 'path', $home_path ) ? $home_path[ 'path' ] : '' );
+            if ( isset( $wp_cache_home_path ) )
+                $sanitized_value =  $wp_cache_home_path;
+            if ( "$home_path" !== "$sanitized_value" )
+                $sanitized_value = $home_path;
+            $sanitized_value = "\"$sanitized_value\"" ;
+            break;
+
+        case 'wp_cache_mobile_browsers':
+            if ( function_exists( "cfmobi_default_browsers" ) ) {
+                $value = cfmobi_default_browsers( "mobile" );
+                $value = array_merge( $value, cfmobi_default_browsers( "touch" ) );
+            } elseif ( function_exists( 'lite_detection_ua_contains' ) ) {
+                $value = explode( '|', lite_detection_ua_contains() );
+            } else {
+                $value = self::MOBILE_BROWSERS;
+            }
+            $sanitized_value = apply_filters( 'cached_mobile_browsers', $value );
+            if ( is_array( $sanitized_value ) ) {
+                $sanitized_value = implode( ', ', $sanitized_value );
+            }
+            $sanitized_value = "\"$sanitized_value\"";
+            break;
+
+        case 'wp_cache_mobile_groups':
```

Comment:
True. It's based on the logic [here](https://github.com/Automattic/wp-super-cache/blob/master/wp-cache.php#L685). Anything that filters the `cached_mobile_groups` should resolve the initial `$value`. Like the `cache_badbehaviour_file` it's here to protect against altered `cached_mobile_groups` settings.